### ISSUE TITLE
error: too many operands for instruction in arch/arm/kernel/relocate_kernel.S

### ISSUE THREAD

**Issue by agners** [ClangBuiltLinux/linux#770]
Trying to compile the kernel in Thumb2 mode with LLVM/clang and integrated assembler leads to the following build error:

```
arch/arm/kernel/relocate_kernel.S:28:12: error: too many operands for instruction                                                                                                                                                                                                                               
 tst r3,#1,0                                                                                                                                                                                                                                                                                                    
           ^                                                                                                                                                                                                                                                                                                    
arch/arm/kernel/relocate_kernel.S:34:12: error: too many operands for instruction                                                                                                                                                                                                                               
 tst r3,#2,0                                                                                                                                                                                                                                                                                                    
           ^                                                                                                                                                                                                                                                                                                    
arch/arm/kernel/relocate_kernel.S:41:12: error: too many operands for instruction                                                                                                                                                                                                                               
 tst r3,#4,0                                                                                                                                                                                                                                                                                                    
           ^                                                                                                                                                                                                                                                                                                    
arch/arm/kernel/relocate_kernel.S:47:12: error: too many operands for instruction                                                                                                                                                                                                                               
 tst r3,#8,0                                                                                                                                                                                                                                                                                                    
           ^
```

**Comment by agners** [comment]
This probably is meant to be the alternative `#<cond>` syntax `#<byte>, #<rot>` valid in ARM mode (however, a pound sign would be still missing). It seems that LLVM/Clang integrated assembler handles it well when compiling a regular ARM kernel, and GCC glosses over it in any case. Fixing it should be fairly trivial and work for all combinations.

**Comment by agners** [comment]
Patch:
https://lore.kernel.org/m/ab67c7c5a1f96af6d22240e57fc27ba766d4193d.1580943526.git.stefan@agner.ch

**Comment by nickdesaulniers** [comment]
Looks like Russell said:
> Please drop it in the patch system, thanks.

**Comment by agners** [comment]
@nickdesaulniers hm, somehow missed that, thanks for reminding me.

Submitted the patch to the tracker: https://www.arm.linux.org.uk/developer/patches/viewpatch.php?id=8962/1

**Comment by nickdesaulniers** [comment]
accepted: https://www.armlinux.org.uk/developer/patches/viewpatch.php?id=8962/1

**Comment by nickdesaulniers** [comment]
fixed: 91274f962e942c23c97ba7e4c7fa04277c533a45
